selfdrive/manager.py

```diff
@@ -478,12 +503,25 @@ def manager_prepare(spinner=None):
   os.chdir(os.path.dirname(os.path.abspath(__file__)))
 
   # Spinner has to start from 70 here
-  total = 100.0 if prebuilt else 50.0
+  total = 100.0 if prebuilt else 70.0
 
+  progress = total
+  prep_failed = False
   for i, p in enumerate(managed_processes):
+    e = prepare_managed_process(p)
+    progress = (100.0 - total) + total * (i + 1) / len(managed_processes)
     if spinner is not None:
-      spinner.update("%d" % ((100.0 - total) + total * (i + 1) / len(managed_processes),))
-    prepare_managed_process(p)
+      if e is None:
+        spinner.update("%d" % progress)
+      else:
+        prep_failed = True
+        for _ in range(10):
+          spinner.update("%d" % progress, format_spinner_error(str(e)))
+          time.sleep(1)
+        break
+
+  if prep_failed:
```

Comment:
Can you reraise the original error when it fails?

Reply:
Done!